File: Array_Lunch/TaskTwo.js
//2
function enter(satisfaction) { 
	return satisfaction[0];
}
function getSpecificCustomers(satisfaction) {
	let filtered = {};
	for(let customer in satisfaction) {
		if(satisfaction[customer] >= 80) filtered[customer] = satisfaction[customer];
	}
	return filtered;
}

function getSatisfactions(filtered) {
	let satisfactions = [];
	
	for(let customer in filtered) {
		satisfactions.push(filtered[customer]);
		}
	return satisfactions;
}

function sortSatisfactions(satisfactions) {
	for(let count = 0; count < satisfactions.length; count++) {
		for(let counter = 0; counter < satisfactions.length; counter++) {	
		if(satisfactions[count] > satisfactions[counter]) {
			let temp = satisfactions[count];
			satisfactions[count] = satisfactions[counter];
			satisfactions[counter] = temp;}
	}
	}
	return satisfactions;
}

function sortFiltered(filtered,satisfactions) {
	sorted = {}
	for(let satisfaction of satisfactions) {
		for(let customer in filtered) {
			if(filtered[customer] == satisfaction) sorted[customer] = satisfaction;
		}
	}
	return sorted;
}
		


let customersSatisfaction = [{
			"Tolu" : 50,
			"Ebuka" : 82,	
			"Martins" : 57,
			"Fola" : 69,
			"Stanley" : 70,
			"Eniife" : 89,
			"Nissi" : 80,
			"Moses" : 100 
}];

customersSatisfaction = enter(customersSatisfaction);
let filtered = getSpecificCustomers(customersSatisfaction);
let satisfactions = getSatisfactions(filtered);
satisfactions = sortSatisfactions(satisfactions);
console.log(sortFiltered(filtered,satisfactions));

File: Array_Lunch/task_two.py
def enter(satisfaction):
	return satisfaction[0]

def getSpecificCustomers(satisfaction):
	filtered = {}
	for key,value in satisfaction.items():
		if value >= 80:
			filtered[key] = value
	return filtered

def getSatisfactions(filtered):
	satisfactions = []	
	for key,value in filtered.items():
		satisfactions.append(value)
	return satisfactions


def sortSatisfactions(satisfactions):
	for count in range(len(satisfactions)): 
		for counter in range(len(satisfactions)):	
			if satisfactions[count] > satisfactions[counter]:
				temp = satisfactions[count]
				satisfactions[count] = satisfactions[counter]
				satisfactions[counter] = temp
	return satisfactions


def sortFiltered(filtered,satisfactions):
	sorted = {}
	for satisfaction in satisfactions:
		for key,value in filtered.items():
			if value == satisfaction: 
				sorted[key] = satisfaction
	return sorted

		


customersSatisfaction = [{
			"Tolu" : 100,
			"Ebuka" : 82,	
			"Martins" : 57,
			"Femi" : 69,
			"Stanley" : 70,
			"Eniife" : 89,
			"Nissi" : 80,
			"Moses" : 100 
}]

satisfaction = enter(customersSatisfaction);
filtered = getSpecificCustomers(satisfaction);
satisfactions = getSatisfactions(filtered);
satisfactions = sortSatisfactions(satisfactions);
print(sortFiltered(filtered,satisfactions));
